walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -121,6 +124,66 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void everyAttendeeIsConsideredWithOptionalAllDay() {
+    // Same as everyAttendeeIsConsidered() but with an extra optional attendee who has an all-day
+    // event. Since they are never available, this should return the same three time slots.
+    //
+    // Events  :       |--A--|     |--B--|
+    //           |==============C==============|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.WHOLE_DAY, Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
```

Comment:
Adding to my above comment, this sort of labeling would be very useful to me on the boolean parameters. I suggest you utilize that strategy here and at every similar occurrence throughout the rest of your code. Just by reading the code, it's not clear to me what those parameters signify.

Reply:
Ahh okay, didn't know that's how you indicate parameter names in cases where it was ambiguous, thanks for the tip! I've added parameter names for the booleans and will keep it in mind for the future.